src/python/pants/backend/python/subsystems/setuptools.py

```diff
@@ -1,16 +1,12 @@
 # Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.subsystems.python_tool_base import PythonToolRequirementsBase
 
 
-class Setuptools(PythonToolBase):
+class Setuptools(PythonToolRequirementsBase):
```

Comment:
This is a breaking change - both `[setuptools].entry_point` and `[setuptools].interpreter_constraints` must be deprecated.

You can keep this subclassing, then implement `register_options()` to register both those options but set `removal_hint` et al.

Reply:
Ah, damn. OK, will fix.